dice_ml/explainer_interfaces/explainer_base.py

```diff
@@ -392,14 +390,13 @@ def do_posthoc_sparsity_enhancement(self, final_cfs_sparse, query_instance, post
 
         cfs_preds_sparse = []
 
-        for cf_ix in range(len(final_cfs_sparse)):
-            current_pred = self.predict_fn_for_sparsity(
-                final_cfs_sparse.iloc[[cf_ix]][self.data_interface.feature_names])
+        for cf_ix in list(final_cfs_sparse.index):
+            current_pred = self.predict_fn_for_sparsity(final_cfs_sparse.loc[[cf_ix]][self.data_interface.feature_names])
```

Comment:
better to use `at` here for speed

Reply:
We can't use `at` here since you can only query a single column using that